src/pykeen/models/unimodal/trans_h.py

```diff
@@ -80,89 +80,25 @@ def __init__(
         """
         super().__init__(
             triples_factory=triples_factory,
+            interaction=TransHInteraction(
+                p=scoring_fct_norm,
+                power_norm=False,
+            ),
             embedding_dim=embedding_dim,
             automatic_memory_optimization=automatic_memory_optimization,
             loss=loss,
             preferred_device=preferred_device,
             random_seed=random_seed,
-            regularizer=regularizer,
+            predict_with_sigmoid=predict_with_sigmoid,
+            embedding_specification=None,
+            relation_embedding_specification=None,
+            second_relation_embedding_specification=EmbeddingSpecification(
+                # Normalise the normal vectors by their l2 norms
+                constrainer=functional.normalize,
+            ),
         )
-
-        self.scoring_fct_norm = scoring_fct_norm
-
-        # embeddings
-        self.normal_vector_embeddings = Embedding.init_with_device(
-            num_embeddings=triples_factory.num_relations,
-            embedding_dim=embedding_dim,
-            device=self.device,
-            # Normalise the normal vectors by their l2 norms
-            constrainer=functional.normalize,
-        )
-
-    def post_parameter_update(self) -> None:  # noqa: D102
-        super().post_parameter_update()
-        self.normal_vector_embeddings.post_parameter_update()
-
-    def _reset_parameters_(self):  # noqa: D102
-        super()._reset_parameters_()
-        self.normal_vector_embeddings.reset_parameters()
-        # TODO: Add initialization
-
-    def regularize_if_necessary(self) -> None:
-        """Update the regularizer's term given some tensors, if regularization is requested."""
-        # As described in [wang2014], all entities and relations are used to compute the regularization term
-        # which enforces the defined soft constraints.
-        super().regularize_if_necessary(
-            self.entity_embeddings(indices=None),
-            self.normal_vector_embeddings(indices=None),  # FIXME
-            self.relation_embeddings(indices=None),
+        self.regularizer = self._instantiate_default_regularizer(
+            entity_embeddings=list(self.entity_representations[0].parameters()).pop(),
```

Comment:
@mberr would you please document what's going on in: `list(self.entity_representations[0].parameters()).pop()`

Reply:
It takes the entity embeddings (=first entity representations). `parameters()` is only an iterator, so I wrapped it in `list(...).pop()` to access the one element contained in this iterator.

Reply:
I am very open towards replacing that by something nicer (list a `first` / `head` function, but I did not find such)

Reply:
So the list of parameters is only one long?

Reply:
Yes, since it is TransH, there should be exactly one entity representation (the entity embedding).